### PR TITLE
Database: validate AWS Account ID

### PR DESCRIPTION
**Pull request by marcoandredinis**
When the AWS Account ID is provided, we must check that it is a valid one.

AWS Account ID is a well documented field: 12-digit string

This check is meant to prevent typos when creating Databases mainly from the Web UI in the context of Teleport Discover.

### CODE REVIEW

File: api/utils/aws/identifiers_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			errCheck:  isBadParamErrFn,
+		},
+		{
+			name:      "invalid chars",
+			accountID: "12345678901A",
+			errCheck:  isBadParamErrFn,
+		},
+		{
+			name:      "invalid chars with emojis",
+			accountID: "12345678901✅",
```
Comment (greedy52) [inline api/utils/aws/identifiers_test.go:63]: 😆

File: api/types/database.go [hunk 2]
```diff
@@ -498,6 +498,13 @@ func (d *DatabaseV3) CheckAndSetDefaults() error {
 		return trace.BadParameter("MySQL ServerVersion can be only set for MySQL database")
 	}
 
+	// Validate AWS Specific configuration
+	if d.Spec.AWS.AccountID != "" {
+		if err := awsutils.IsValidAccountID(d.Spec.AWS.AccountID); err != nil {
+			return trace.BadParameter("invalid AWS Account ID: %v", err)
+		}
+	}
+
```
Comment (greedy52) [inline api/types/database.go:507]: Some of the URL parsers below extract account IDs from the URI, so I think we should validate this after the switch.

File: api/utils/aws/identifiers.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+// IsValidAccountID checks whether the accountID is a valid AWS Account ID
+//
+// https://docs.aws.amazon.com/accounts/latest/reference/manage-acct-identifiers.html
+func IsValidAccountID(accountID string) error {
+	if len(accountID) != 12 {
+		return trace.BadParameter("must be 12-digit")
+	}
+	for _, d := range accountID {
+		if !unicode.IsDigit(d) {
```
Comment (greedy52) [inline api/utils/aws/identifiers.go:33]: Wouldn't [`unicode.IsDigit`](https://pkg.go.dev/unicode#IsDigit) also test true for unicode digit? I believe we only want ASCII digits?

**Comment by greedy52** [comment]
Thanks for updating all tests for this!

File: api/types/database.go (hunk 2 above)
Comment (marcoandredinis) [reply api/types/database.go:507]: :+1:

File: api/utils/aws/identifiers.go (hunk 3 above)
Comment (marcoandredinis) [reply api/utils/aws/identifiers.go:33]: I think so, yeah

I guess I'll just copy the first half of the function to here as there doesn't seem to be a function that validates just ASCII digits

**Review by greedy52 — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]
